### ISSUE TITLE
[v4] The extractor can't handle `TSTypeParameterInstantiation` types

### ISSUE THREAD

**Issue by zvictor** [lingui/js-lingui#1616]
**Describe the bug**
The extractor can't handle `TSTypeParameterInstantiation` types.

**To Reproduce**
```haskell
git clone https://github.com/zvictor/lingui-macro-bug ; cd lingui-macro-bug
pnpm i
pnpm run extract
```

Running the above, you will see `Cannot process file /private/tmp/lingui-macro-bug/src/PlasmicHomepage.tsx /private/tmp/lingui-macro-bug/src/PlasmicHomepage.tsx: @lingui/macro: Cannot destructure property 'attributes' of 'path.node.openingElement' as it is undefined`.

Altering the code of `stripMacroAttributes` slightly to set `attributes` as `[]` whenever `openingElement` is undefined leads to the error `Property value of ObjectProperty expected node to be of a type ["Expression","PatternLike"] but instead got "TSTypeParameterInstantiation"`.

**Additional context**
The code of `src/PlasmicHomepage.tsx` is complex because it's auto generated by https://plasmic.app, making it harder to manipulate.

**Versions**
```json
{
  "devDependencies": {
    "@babel/core": "^7.21.4",
    "@lingui/cli": "4.0.0-next.8",
    "@lingui/macro": "4.0.0-next.8",
    "babel-plugin-macros": "^3.1.0"
  },
  "dependencies": {
    "@lingui/react": "4.0.0-next.8",
    "@lingui/swc-plugin": "4.0.0-next.3"
  }
}
```

**Comment by zvictor** [comment]
I have no prior experience with babel and getting to this reproducible already took me many hours 🥹

Without gpt4's help i couldn't even understand the error 😅.

Thus i'm afraid this is already the minimal reproducible i can get to. I'm sorry.

**Comment by thekip** [comment]
Take the file which is causing an issue, and remove code in in the file on binary search manner. Until you narrow down it to the construction which is failing the extractor.

**Comment by zvictor** [comment]
I can confirm that the fix on #1617 solves the issue. The original `PlasmicHomepage.tsx` file runs smoothly with the change 👏
